src/generator/operations.ts

```diff
@@ -1039,14 +1043,19 @@ function generateReturnsInfo(op: Operation, forHandler: boolean): string[] {
   if (isMultiRespOperation(op)) {
     returnType = 'interface{}';
   } else {
-    const firstResp = op.responses![0];
-    // must check pageable first as all pageable operations are also schema responses
-    if (!forHandler && isPageableOperation(op)) {
+    const firstResp = <SchemaResponse>op.responses![0];
+    // must check pageable first as all pageable operations are also schema responses, 
+    // but LRO operations that return a pager are an exception and need to return LRO specific
+    // responses
+    if (!forHandler && isPageableOperation(op) && !isLROOperation(op)) {
       returnType = op.language.go!.pageableType.name;
     } else if (isSchemaResponse(firstResp)) {
       returnType = '*' + firstResp.schema.language.go!.responseType.name;
+      if (isLROOperation(op) || (isLROOperation(op) && forHandler)) {
```

Comment:
The second half of the OR clause will never be true.

Reply:
Oops, I thought I had removed this. Will do

Reply:
Done